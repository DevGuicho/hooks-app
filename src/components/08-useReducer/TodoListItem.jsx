import React from "react";

const TodoListItem = ({ todo, i, handleDelete, handleToggle }) => {
  return (
    <tr className={todo.done ? "table-success" : ""} key={todo.id}>
      <th scope="row">{i + 1}</th>
      <td id="desc">{todo.desc}</td>
      <td>
        <button
          type="button"
          className="btn btn-danger"
          onClick={() => handleDelete(todo.id)}
        >
          Borrar
        </button>
      </td>
      <td>
        <div className="form-check">
          <input
            className="form-check-input"
            type="checkbox"
            value=""
            id="flexCheckChecked"
            onChange={() => handleToggle(todo.id)}
            checked={todo.done}
          />
        </div>
      </td>
    </tr>
  );
};

export default TodoListItem;
