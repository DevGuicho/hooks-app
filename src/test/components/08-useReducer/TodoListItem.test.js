import { shallow } from "enzyme";
import TodoListItem from "../../../components/08-useReducer/TodoListItem";
import { demoTodos } from "../../fixtures/demoTodos";

describe("Pruebas en <TodoListItem />", () => {
  const handleDelete = jest.fn();
  const handleToggle = jest.fn();

  const wrapper = shallow(
    <TodoListItem
      todo={demoTodos[0]}
      i={0}
      handleDelete={handleDelete}
      handleToggle={handleToggle}
    />
  );
  test("debe mostrar el componente correctamente", () => {
    expect(wrapper).toMatchSnapshot();
  });
  test("debe llamar la funcion handleDelete", () => {
    wrapper.find("button").simulate("click");
    expect(handleDelete).toHaveBeenCalledWith(demoTodos[0].id);
  });
  test("debe llamar la funcion handleToggle", () => {
    wrapper.find("input").simulate("change");
    expect(handleToggle).toHaveBeenCalledWith(demoTodos[0].id);
  });
  test("debe mostrar correctamente el texto", () => {
    expect(wrapper.find("#desc").text().trim()).toBe(demoTodos[0].desc);
  });
  test("debe de tener la clase complete si el todo.done === true", () => {
    const todo = demoTodos[0];
    todo.done = true;
    const wrapper = shallow(
      <TodoListItem
        todo={todo}
        i={0}
        handleDelete={handleDelete}
        handleToggle={handleToggle}
      />
    );
    expect(wrapper.find("tr").hasClass("table-success")).toBe(true);
  });
});
