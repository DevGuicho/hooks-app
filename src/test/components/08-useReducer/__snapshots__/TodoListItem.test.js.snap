// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <TodoListItem /> debe mostrar el componente correctamente 1`] = `
<tr
  className=""
  key="1"
>
  <th
    scope="row"
  >
    1
  </th>
  <td
    id="desc"
  >
    Hola Mundo
  </td>
  <td>
    <button
      className="btn btn-danger"
      onClick={[Function]}
      type="button"
    >
      Borrar
    </button>
  </td>
  <td>
    <div
      className="form-check"
    >
      <input
        checked={false}
        className="form-check-input"
        id="flexCheckChecked"
        onChange={[Function]}
        type="checkbox"
        value=""
      />
    </div>
  </td>
</tr>
`;
